### ISSUE TITLE
Support plotting quantiles of the data distribution with lineplot

### ISSUE THREAD

**Issue by shoyer** [mwaskom/seaborn#1501]
`seaborn.lineplot` makes it easy to plot either a confidence intervals around the estimator, or the standard deviation of the data. It would be nice to directly support plotting quantiles of the data distribution instead, rather than only the standard deviation.

Maybe it would make sense to have a separate key-word argument for controlling whether `ci` refers to the estimator or data distribution? e.g., `ci_source='estimator'` vs `ci_source='data'`? Then `ci` would only indicate how large of a confidence interval to plot, not the source.

**Comment by shoyer** [comment]
Looking more carefully at Seaborn's API, I see that the `ci` argument is also used with the same meaning in `pointplot` and `barplot`. So certainly caution is warranted in updating this API, given that these are *not* new functions.

My normal inclination would be to support this sort of thing by adding configuration objects (e.g., `ci=EstimateCI(alpha=0.05)` or `ci=DataCI(standard_deviations=2)`), but this would be an unusual pattern for an external API in Seaborn.

Maybe an `dict` would make more sense for Seaborn, e.g., `ci=dict(source='estimate', standard_deviations=2)`? If we're OK with fully explicit names, we could start with fields:
- `source='data'` or `source='estimate`
- `standard_deviations` or `confidence_level` (mutually exclusive) for indicating the number of standard deviations or the confidence level

**Comment by mwaskom** [comment]
> Looking more carefully at Seaborn's API, I see that the ci argument is also used with the same meaning in pointplot and barplot.

Also in `regplot`. And I got lazy when implementing `ci="sd"` and just duplicated the logic everywhere instead of writing so more abstract way of doing things.

Another reason I've been resistant to overloading the `ci` parameter when this came up in the past is terminological. It doesn't *really* make sense to say "the confidence interval of the data" (I think technically this is a "predictive interval?") or "the confidence interval is two standard deviations". A better forward-looking name for this would have been derived from "error bar" that could take a simple tuple spec (e.g. `ebar=("ci", 95)`).

> My normal inclination would be to support this sort of thing by adding configuration objects (e.g., ci=EstimateCI(alpha=0.05) or ci=DataCI(standard_deviations=2)), but this would be an unusual pattern for an external API in seaborn.

Yeah I've also thought about something like this, especially when we were discussing adding more advanced bootstrapping algorithms that require a few more parameters, but as you say it would be unlike the way anything else works in seaborn.

**Comment by cypressf** [comment]
For those of us that want quantile shading now, is there a recommended way to implement it on our own by overwriting internal seaborn methods or using some lower-level matplotlib functions?

**Comment by mwaskom** [comment]
@shoyer this is going to happen but there's some uncertainty about what to call this kind of interval (see #2403). Naively I would call it a "percentile interval", but google searches for that are dominated by discussion of the bootstrap. Which makes sense, but is there a different term you are aware of for when you are constructing a nonparametric interval from observations, rather than bootstrap replicates?

**Comment by mwaskom** [comment]
Closed by #2407.

e.g. to show the 25-75 percentile interval:

```python
fmri = sns.load_dataset("fmri")
sns.lineplot(data=fmri, x="timepoint", y="signal", errorbar=("pi", 50))
```
